examples/linear_model/plot_poisson_regression_non_normal_loss.py

```diff
@@ -352,81 +394,102 @@ def _mean_frequency_by_risk_group(y_true, y_pred, sample_weight=None,
     return bin_centers, y_true_bin, y_pred_bin
 
 
-fig, ax = plt.subplots(nrows=1, ncols=3, figsize=(12, 3.5))
+print(f"Actual number of claims: {df_test['ClaimNb'].sum()}")
+fig, ax = plt.subplots(nrows=2, ncols=2, figsize=(12, 8))
 plt.subplots_adjust(wspace=0.3)
 
-for axi, model in zip(ax, [ridge, poisson, rf]):
+for axi, model in zip(ax.ravel(), [ridge_glm, poisson_glm, poisson_gbrt,
+                                   dummy]):
     y_pred = model.predict(df_test)
-
+    y_true = df_test["Frequency"].values
+    exposure = df_test["Exposure"].values
     q, y_true_seg, y_pred_seg = _mean_frequency_by_risk_group(
-        df_test["Frequency"].values,
-        y_pred,
-        sample_weight=df_test["Exposure"].values,
-        n_bins=10)
+        y_true, y_pred, sample_weight=exposure, n_bins=10)
+
+    # Name of the model after the estimator used in the last step of the
+    # pipeline.
+    print(f"Predicted number of claims by {model[-1]}: "
+          f"{np.sum(y_pred * exposure):.1f}")
 
-    axi.plot(q, y_pred_seg, marker='o', linestyle="-", label="predictions")
-    axi.plot(q, y_true_seg, marker='x', linestyle="--", label="observations")
+    axi.plot(q, y_pred_seg, marker='x', linestyle="--", label="predictions")
+    axi.plot(q, y_true_seg, marker='o', linestyle="--", label="observations")
     axi.set_xlim(0, 1.0)
-    axi.set_ylim(0, 0.6)
+    axi.set_ylim(0, 0.5)
     axi.set(
-        title=model[-1].__class__.__name__,
+        title=model[-1],
         xlabel='Fraction of samples sorted by y_pred',
         ylabel='Mean Frequency (y_pred)'
     )
     axi.legend()
 plt.tight_layout()
 
-##############################################################################
-# The ``Ridge`` regression model can predict very low expected frequencies
-# that do not match the data. It can therefore severly under-estimate the risk
-# for some policyholders.
+###############################################################################
+# The dummy regression model predicts a constant frequency. This model is not
+# attribute the same tied rank to all samples but is none-the-less globally
```

Comment:
should "is not" be "does not"?

Though I'm not sure I understand the plot for the dummy: if it always predicts the mean, why do we have multiple dots on the x-axis? I would have assumed the plot should just be a single dot?

Reply:
We bin by quantile so the content of each bin is independent of y hat (since y hat is constant) and therefore uniformly random.

I agree it's an edge case. But it really high light that this notion of probabilistic calibration and ranking power are quite independent of one another.